webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -749,30 +792,40 @@ private static PdfPTable createTable(PdfWriter docWriter, Document doc, float si
             l.log(Level.INFO, "", e);
         }
 
+        //check if this puzzle should have highlighted scrambles or not
+        boolean highlight = scrambler.shouldHighlightAlternatingLines();
+
+        int maxLinesPerScramble = 0;
+        for(String scramble : scrambles) {
+            String paddedScramble = oneLine ? scramble : padTurnsUniformly(scramble, " ");
+            LinkedList<Chunk> lineChunks = splitScrambleToLineChunks(paddedScramble, scrambleFont, availableScrambleWidth);
+            if(lineChunks.size() > maxLinesPerScramble) {
+                maxLinesPerScramble = lineChunks.size();
+            }
+        }
         for(int i = 0; i < scrambles.length; i++) {
             String scramble = scrambles[i];
+            String paddedScramble = oneLine ? scramble : padTurnsUniformly(scramble, " ");
             Chunk ch = new Chunk(scrambleNumberPrefix + (i+1) + ".");
             PdfPCell nthscramble = new PdfPCell(new Paragraph(ch));
             nthscramble.setVerticalAlignment(PdfPCell.ALIGN_MIDDLE);
             table.addCell(nthscramble);
 
-            String paddedScramble;
-            if(oneLine) {
-                paddedScramble = scramble;
-            } else {
-                paddedScramble = padTurnsUniformly(scramble, " ");
-            }
-            Chunk scrambleChunk = new Chunk(paddedScramble);
-            if(paddedScramble.indexOf("\n") >= 0) {
-                // If the scramble contains newlines, then we *only* allow wrapping at the
-                // newlines.
-                scrambleChunk.setSplitCharacter(SPLIT_ON_NEWLINES);
-            } else {
-                scrambleChunk.setSplitCharacter(SPLIT_ON_SPACES);
+            Phrase scramblePhrase = new Phrase();
+            boolean oddLine = false;
+            LinkedList<Chunk> lineChunks = splitScrambleToLineChunks(paddedScramble, scrambleFont, availableScrambleWidth);
+            for(Chunk lineChunk : lineChunks) {
+                oddLine = !oddLine;
+                if(highlight && oddLine) {
+                    lineChunk.setBackground(new BaseColor(220, 220, 250));
```

Comment:
Lets pull this color out into a `private static final` variable at the top of this file.